File: ndb/greedy/example3_2.py
n, m, k = map(int, input().split())
arr = list(map(int, input().split()))
arr.sort()

answer = 0
max_arr = [arr[len(arr)-1], arr[len(arr)-2]]

count = 0
for i in range(m):
    if count >= k:
        answer += max_arr[1]
        count = 0
        continue
    answer += max_arr[0]
    count += 1

print(answer)

File: ndb/greedy/example3_2.js
function solution(n, m, k, arr) {
  let answer = 0;
  let count = 0;
  arr.sort((a, b) => b - a);

  const first = arr[0];
  const second = arr[1];

  for (let i = 0; i < m; i++) {
    if (count < k) {
      answer += first;
      count += 1;
    } else {
      answer += second;
      count = 0;
    }
  }
  return answer;
}

console.log(solution(5, 8, 3, [2, 4, 5, 4, 6]));
